neofs-cli panic: runtime error on attempt to set eACL for specific key

1) eACL example:
```
{
  "records": [
    {
      "operation": "GET",
      "action": "ALLOW",
      "targets": [
        {
          "keys": [ "NSNKmYXGM6TUH4AjAbtC2afxbJMV87XdDT" ]
        }
      ]
    }
  ]
}


```

2) Try to set eACL:
```
neofs-cli --rpc-endpoint s01.neofs.devenv:8080 --key KzKYJ7sa9SswhSPMiSLdEMZZKEZ3AUJJRo6dCbxRKqE7tDH4mPaP  container set-eacl --cid 4uSfsDmfK6hPKSfDwQLNkagcHDiwKwsY3rnp1keFa4wo --table robot/resources/lib/eacl/eacl_encoded_allow_all_pubkey
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x932ce1]

goroutine 1 [running]:
github.com/nspcc-dev/neofs-api-go/pkg/acl/eacl.NewTargetFromV2(0xc00027f040, 0x0)
	github.com/nspcc-dev/neofs-api-go@v1.20.3-0.20201123111542-5ee500bb43c6/pkg/acl/eacl/target.go:74 +0x141
github.com/nspcc-dev/neofs-api-go/pkg/acl/eacl.NewRecordFromV2(0xc000262c00, 0x0)
	github.com/nspcc-dev/neofs-api-go@v1.20.3-0.20201123111542-5ee500bb43c6/pkg/acl/eacl/record.go:171 +0x144
github.com/nspcc-dev/neofs-api-go/pkg/acl/eacl.NewTableFromV2(0xc0000f3b78, 0xc0002ce000)
	github.com/nspcc-dev/neofs-api-go@v1.20.3-0.20201123111542-5ee500bb43c6/pkg/acl/eacl/table.go:121 +0x1c1
github.com/nspcc-dev/neofs-api-go/pkg/acl/eacl.(*Table).UnmarshalJSON(0xc0002944b0, 0xc0002ce000, 0xc1, 0x2c1, 0x2c1, 0x0)
	github.com/nspcc-dev/neofs-api-go@v1.20.3-0.20201123111542-5ee500bb43c6/pkg/acl/eacl/table.go:166 +0x83
github.com/nspcc-dev/neofs-node/cmd/neofs-cli/modules.parseEACL(0x7fff2b74729f, 0x36, 0xc00027e8e0, 0x0, 0x0)
	github.com/nspcc-dev/neofs-node/cmd/neofs-cli/modules/container.go:688 +0x258
github.com/nspcc-dev/neofs-node/cmd/neofs-cli/modules.glob..func10(0x129e3c0, 0xc00028e180, 0x0, 0x8, 0x0, 0x0)
	github.com/nspcc-dev/neofs-node/cmd/neofs-cli/modules/container.go:391 +0xbd
github.com/spf13/cobra.(*Command).execute(0x129e3c0, 0xc00028e080, 0x8, 0x8, 0x129e3c0, 0xc00028e080)
	github.com/spf13/cobra@v1.0.0/command.go:842 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0x129cec0, 0x0, 0x0, 0x0)
	github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.0.0/command.go:887
github.com/nspcc-dev/neofs-node/cmd/neofs-cli/modules.Execute()
	github.com/nspcc-dev/neofs-node/cmd/neofs-cli/modules/root.go:55 +0x2d
main.main()
	github.com/nspcc-dev/neofs-node/cmd/neofs-cli/main.go:6 +0x20
```

--- a/go.mod
+++ b/go.mod
@@ -14,7 +14,7 @@ require (
 	github.com/multiformats/go-multiaddr-net v0.1.2 // v0.1.1 => v0.1.2
 	github.com/multiformats/go-multihash v0.0.13 // indirect
 	github.com/nspcc-dev/neo-go v0.91.1-pre.0.20201030072836-71216865717b
-	github.com/nspcc-dev/neofs-api-go v1.20.3-0.20201126102634-2a94fdc5e709
+	github.com/nspcc-dev/neofs-api-go v1.20.3-0.20201201071132-c01024b5531a
 	github.com/nspcc-dev/neofs-crypto v0.3.0
 	github.com/nspcc-dev/tzhash v1.4.0
 	github.com/panjf2000/ants/v2 v2.3.0

--- a/pkg/services/object/acl/eacl/v2/eacl_test.go
+++ b/pkg/services/object/acl/eacl/v2/eacl_test.go
@@ -124,12 +124,12 @@ func TestHeadRequest(t *testing.T) {
 
 	senderKey := test.DecodeKey(-1).PublicKey
 
-	r := new(eacl.Record)
+	r := eacl.NewRecord()
 	r.SetOperation(eacl.OperationHead)
 	r.SetAction(eacl.ActionDeny)
 	r.AddFilter(eacl.HeaderFromObject, eacl.MatchStringEqual, attrKey, attrVal)
 	r.AddFilter(eacl.HeaderFromRequest, eacl.MatchStringEqual, xKey, xVal)
-	r.AddTarget(eacl.RoleUnknown, senderKey)
+	eacl.AddFormedTarget(r, eacl.RoleUnknown, senderKey)
 
 	table.AddRecord(r)
 

--- a/pkg/services/object/acl/eacl/validator.go
+++ b/pkg/services/object/acl/eacl/validator.go
@@ -5,7 +5,6 @@ import (
 	"errors"
 
 	"github.com/nspcc-dev/neofs-api-go/pkg/acl/eacl"
-	crypto "github.com/nspcc-dev/neofs-crypto"
 	"github.com/nspcc-dev/neofs-node/pkg/core/container"
 	"github.com/nspcc-dev/neofs-node/pkg/util/logger"
 	"go.uber.org/zap"
@@ -160,8 +159,8 @@ func matchFilters(hdrSrc TypedHeaderSource, filters []*eacl.Filter) int {
 func targetMatches(unit *ValidationUnit, record *eacl.Record) bool {
 	for _, target := range record.Targets() {
 		// check public key match
-		for _, key := range target.Keys() {
-			if bytes.Equal(crypto.MarshalPublicKey(&key), unit.key) {
+		for _, key := range target.BinaryKeys() {
+			if bytes.Equal(key, unit.key) {
 				return true
 			}
 		}

